### ISSUE TITLE
efs-provisioner

### ISSUE THREAD

**Issue by pierluigilenoci** [kubernetes-sigs/sig-storage-lib-external-provisioner#91]
I am opening this ticket because I wanted information about efs-provisioner. The repo that contained it has been archived https://github.com/kubernetes-retired/external-storage/ and I have not found any information on the future of the project. Can you help me? Maybe @wongma7 ?

Ref: https://github.com/kubernetes-retired/external-storage/tree/master/aws/efs

**Comment by wongma7** [comment]
I recommend moving to https://github.com/kubernetes-sigs/nfs-subdir-external-provisioner . It is a generic and less opinionated ( does not try to validate stuff) version of the efs provisioner

**Comment by pierluigilenoci** [comment]
I'll try this new provisioner. Thank you.

**Comment by chadlwilson** [comment]
Hi @wongma7 - do you know of a migration guide/suggestion/procedure somewhere for transitioning from the `efs-provisioner` to the `nfs-subdir-external-provisioner` for those with existing production systems and PVCs/PVs?

I'm trying to assess what we might be getting into with such a migration to a supported option (c.f switching to the EFS CSI driver, which is obviously an alternative).

Do you have any useful experience to share from your perspective @pierluigilenoci ?

**Comment by pierluigilenoci** [comment]
@chadlwilson I'm sorry but I haven't started migrating to the new provider yet so my experience is limited. I am waiting for the migration of the chart before starting to do the job once. Ref: https://github.com/helm/charts/issues/21103 

@verwilst is the chart maintainer, maybe he can help you better.

**Comment by chadlwilson** [comment]
I had forgotten about this issue, however if anyone else comes across it, in our case we found the migration to `nfs-subdir-external-provisioner` relatively painless. Needed to create new PVCs, but were able to do so on the same underlying EFS volume, migrate apps across and then remove the old PVCs and PVs before uninstalling the `efs-provisioner`.

In our case we didn't have data stored here that we couldn't afford to lose, making it easier.

Couple of differences to note
- no longer requires AWS creds (so need for kube2iam etc)
- I found a hack to allow the new provisioner to 'adopt' the previous volumes under the efs storage class which seemed to work fine, BUT it involved losing control of configuration values in the storage class (since these are immutable) and did not seem a good idea so we created a new storage class and new volumes in the end.
- the new provisioner is distroless so better from a security perspective
- new provisioner supports archive-on-delete which has is a different default behaviour to the old one

A hacky migration strategy for those who cannot lose data may be to exec into the old provisioner and copy files from the old volume subdir to the new subdir, being careful about permissions (if the provisioner shares EFS volume it can see the subdirs created by the other provisioner, and it has a shell in the container).
